### PR TITLE
Add invalidBeforeDate and invalidAfterDate with a textStyle that defa…

### PR DESCRIPTION
**Pull request by edwintoh-work**
…ults to 0.2 opacity and overrides other styles.

Invalid Dates after 6/10:
image

Invalid Dates before 6/10:
image

### CODE REVIEW

File: components/Calendar.js [hunk 1]
```diff
@@ -238,17 +245,21 @@ export default class Calendar extends Component {
...
 
       if (dayIndex >= 0 && dayIndex < argDaysCount) {
         days.push((
           <Day
+             isInvalid={isInvalid}
              startOfMonth={startOfArgMoment}
              isWeekend={isoWeekday === 0 || isoWeekday === 6}
              key={`${renderIndex}`}
-             onPress={() => {
+             onPress={!isInvalid ? () => {
```
Comment (sarahatwork) [inline components/Calendar.js]: When a day is invalid, could we pass `undefined` instead and make `onPress` an optional property?

File: components/Day.js [hunk 2]
```diff
@@ -18,13 +18,22 @@ export default class Day extends Component {
...
+    isInvalid: PropTypes.bool,
     isSelected: PropTypes.bool,
     isToday: PropTypes.bool,
     isWeekend: PropTypes.bool,
     onPress: PropTypes.func,
     showEventIndicators: PropTypes.bool,
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    if (!this.props.isInvalid || !nextProps.isInvalid) {
```
Comment (sarahatwork) [inline components/Day.js]: Nitpick - we could simplify this to `return !this.props.isInvalid || !nextProps.isInvalid`

Comment (sarahatwork) [inline components/Day.js]: Also, can you clarify why we need this `shouldComponentUpdate`?

Comment (edwintoh-work) [inline components/Day.js]: I noticed the JS thread drops a lot of frames when we were selecting a date (this happens on the original version), so I wanted to make a small perf update to this component. It didn't really help so I think I'll take it out.

**Pull request merged**